File: src/main/java/cn/evolvefield/mirai/onebot/config/PluginConfig.java
package cn.evolvefield.mirai.onebot.config;

import lombok.Data;
import lombok.NoArgsConstructor;
import net.mamoe.mirai.console.data.Value;
import net.mamoe.mirai.console.data.java.JavaAutoSavePluginConfig;

import java.util.HashMap;
import java.util.Map;

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/10/3 20:25
 * Version: 1.0
 */
public class PluginConfig extends JavaAutoSavePluginConfig {
    public static final PluginConfig INSTANCE = new PluginConfig();

    public PluginConfig() {
        super("settings");

    }

    public final Value<Map<String, BotConfig>> bots = typedValue("bots",
            createKType(Map.class, createKType(String.class), createKType(BotConfig.class)),
            new HashMap<>() {{
                put("123454321", new BotConfig());
            }}
    );

    @Data
    public static class DBConfig{
        boolean enable = true;
    }



    @Data
    @NoArgsConstructor
    public static class HeartbeatSettings{
        boolean enable = false;
        long interval = 1500L;
    }

    @Data
    @NoArgsConstructor
    public static class WebsocketServerSettings{
        boolean enable = false;
        String postMessageFormat = "string";
        String wsHost = "0.0.0.0";
        int wsPort = 8080;
        String accessToken = "";
    }
}

File: src/main/kotlin/cn/evole/onebot/mirai/config/PluginConfig.kt
package cn.evole.onebot.mirai.config

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import net.mamoe.mirai.console.data.AutoSavePluginConfig
import net.mamoe.mirai.console.data.value

object PluginConfig : AutoSavePluginConfig("setting"){
    val bots: MutableMap<String, BotConfig>? by value(mutableMapOf("12345654321" to BotConfig()))

    @Serializable
    data class BotConfig(
        var cacheImage: Boolean = false,
        var cacheRecord: Boolean = false,
        var heartbeat: HeartbeatConfig = HeartbeatConfig(),
        var http: HTTPConfig = HTTPConfig(),
        @SerialName("ws_reverse")
        var wsReverse: MutableList<WSReverseConfig> = mutableListOf(WSReverseConfig()),
        var ws: WSConfig = WSConfig()
    )

    @Serializable
    data class DBSettings(
        var enable: Boolean = true,
    )

    @Serializable
    data class HeartbeatConfig(
        var enable: Boolean = false,
        var interval: Long = 1500L
    )

    @Serializable
    data class HTTPConfig(
        var enable: Boolean = false,
        var host: String = "0.0.0.0",
        var port: Int = 6700,
        var accessToken: String = "",
        var postUrl: String = "",
        var postMessageFormat: String = "string",
        var secret: String = "",
        var timeout: Long = 0L
    )

    @Serializable
    data class WSReverseConfig(
        var enable: Boolean = false,
        var postMessageFormat: String = "string",
        var reverseHost: String = "127.0.0.1",
        var reversePort: Int = 8090,
        var accessToken: String = "",
        var reversePath: String = "/ws",
        var reverseApiPath: String = "/api",
        var reverseEventPath: String = "/event",
        var useUniversal: Boolean = true,
        var useTLS: Boolean = false,
        var reconnectInterval: Long = 3000L,
    )

    @Serializable
    data class WSConfig(
        var enable: Boolean = false,
        var postMessageFormat: String = "string",
        var wsHost: String = "0.0.0.0",
        var wsPort: Int = 8080,
        var accessToken: String = ""
    )
}